src/webapi/html_elements/slot.rs

```diff
@@ -0,0 +1,77 @@
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webapi::event_target::{IEventTarget, EventTarget};
+use webapi::node::{INode, Node};
+use webapi::element::{IElement, Element};
+use webapi::html_element::{IHtmlElement, HtmlElement};
+
+/// The HTML `<slot>` element represents a placeholder inside a web component that
+/// you can fill with your own markup, which lets you create separate DOM trees and
+/// present them together.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot)
+// https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "HTMLSlotElement")]
+#[reference(subclass_of(EventTarget, Node, Element, HtmlElement))]
+pub struct SlotElement( Reference );
+
+impl IEventTarget for SlotElement {}
+impl INode for SlotElement {}
+impl IElement for SlotElement {}
+impl IHtmlElement for SlotElement {}
+
+impl SlotElement {
+    /// The slot's name
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/name)
+    // https://html.spec.whatwg.org/multipage/scripting.html#attr-slot-name
+    #[inline]
+    pub fn name ( &self ) -> String {
+        js! (
+            return @{self}.name;
+        ).try_into().unwrap()
+    }
+
+    /// Setter of name.
+    #[inline]
+    pub fn set_name<S: AsRef<str>>( &self, new_name: S ) {
+        js! ( @(no_return)
+            @{self}.name = @{new_name.as_ref()};
+        );
+    }
+
+    /// Returns slot's assigned nodes.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/assignedNodes)
+    // https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element:dom-slot-assignednodes
+    pub fn assigned_nodes( &self, flatten: bool ) -> Vec<Node> {
```

Comment:
Rather than take a `bool` parameter I'd probably split this into two methods, say, `assigned_nodes` and `flattened_assigned_nodes`? (The second name is a little long though; if you can think of a better one feel free to suggest it.) Ditto for `assigned_elements`.

Reply:
How about a enum (like `webapi::node::CloneKind`)?

```rust
enum ChildrenArrangement { // I cannot come up with a better name.
    Flattened,
    Nested,
}

pub fn assigned_nodes( &self, a: ChildrenArrangement ) -> ... {
```

Also, should the return type be `NodeList`?

Reply:
> Also, should the return type be `NodeList`?

In the specs it says that it's a `sequence<Node>`, which is different than `NodeList`, so I don't think so.

> How about a enum (like `webapi::node::CloneKind`)?

Well, that would work too, but that's another extra type which you have to import and remember.

I don't have a strong opinion either way though.